Provide meaningful error messages for MP Reactive Messaging

**Is your feature request related to a problem? Please describe.**
If I deploy a MP reactive messaging app in Liberty and something goes wrong, I get these very generic lines in my console.log:
```
[ERROR   ] Unable to create the publisher or subscriber during initialization
java.lang.reflect.InvocationTargetException
[ERROR   ] CWWKZ0002E: An exception occurred while starting the application kitchen. The exception message was: com.ibm.ws.container.service.state.StateChangeException: org.jboss.weld.exceptions.DeploymentException
[AUDIT   ] CWWKF0012I: The server installed the following features: [cdi-2.0, concurrent-1.0, jaxrs-2.1, jaxrsClient-2.1, jndi-1.0, json-1.0, jsonb-1.0, jsonp-1.1, mpConfig-1.3, mpHealth-2.1, mpReactiveMessaging-1.0, mpReactiveStreams-1.0, servlet-4.0].
```

This gets even worse when running in a Docker container (in production or using MicroShed Testing). When I dug out the stack trace for this particular error message, I get the full stack trace:
```
[3/11/20 20:43:25:614 GMT] 0000002c io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory E Unable to create the publisher or subscriber during initialization
com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterException: java.lang.reflect.InvocationTargetException
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory.getInstance(KafkaAdapterFactory.java:167)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory.getInstance(KafkaAdapterFactory.java:156)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory.newKafkaConsumer(KafkaAdapterFactory.java:95)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.AdapterFactoryImpl$Proxy$_$$_WeldClientProxy.newKafkaConsumer(Unknown Source)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.KafkaIncomingConnector.getPublisherBuilder(KafkaIncomingConnector.java:115)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.KafkaIncomingConnector$Proxy$_$$_WeldClientProxy.getPublisherBuilder(Unknown Source)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.createPublisherBuilder(ConfiguredChannelFactory.java:145)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.lambda$register$4(ConfiguredChannelFactory.java:123)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory$$Lambda$100/00000000F86FECB0.accept(Unknown Source)
	at java.util.HashMap.forEach(HashMap.java:1289)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.register(ConfiguredChannelFactory.java:123)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.initialize(ConfiguredChannelFactory.java:118)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory$Proxy$_$$_WeldClientProxy.initialize(Unknown Source)
	at io.smallrye.reactive.messaging.extension.MediatorManager$$Lambda$98/00000000F86FE500.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:647)
	at io.smallrye.reactive.messaging.extension.MediatorManager.initializeAndRun(MediatorManager.java:132)
	at io.smallrye.reactive.messaging.extension.MediatorManager$Proxy$_$$_WeldClientProxy.initializeAndRun(Unknown Source)
	at io.smallrye.reactive.messaging.extension.ReactiveMessagingExtension.afterDeploymentValidation(ReactiveMessagingExtension.java:86)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
	at org.jboss.weld.injection.MethodInvocationStrategy$SpecialParamPlusBeanManagerStrategy.invoke(MethodInvocationStrategy.java:187)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:330)
	at org.jboss.weld.event.ExtensionObserverMethodImpl.sendEvent(ExtensionObserverMethodImpl.java:123)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:308)
	at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:286)
	at javax.enterprise.inject.spi.ObserverMethod.notify(ObserverMethod.java:124)
	at org.jboss.weld.util.Observers.notify(Observers.java:166)
	at org.jboss.weld.event.ObserverNotifier.notifySyncObservers(ObserverNotifier.java:285)
	at org.jboss.weld.event.ObserverNotifier.notify(ObserverNotifier.java:273)
	at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:177)
	at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:171)
	at org.jboss.weld.bootstrap.events.AbstractContainerEvent.fire(AbstractContainerEvent.java:53)
	at org.jboss.weld.bootstrap.events.AbstractDeploymentContainerEvent.fire(AbstractDeploymentContainerEvent.java:35)
	at org.jboss.weld.bootstrap.events.AfterDeploymentValidationImpl.fire(AfterDeploymentValidationImpl.java:28)
	at org.jboss.weld.bootstrap.WeldStartup.validateBeans(WeldStartup.java:505)
	at org.jboss.weld.bootstrap.WeldBootstrap.validateBeans(WeldBootstrap.java:93)
	at com.ibm.ws.cdi.impl.CDIContainerImpl.startInitialization(CDIContainerImpl.java:157)
	at com.ibm.ws.cdi.liberty.CDIRuntimeImpl.applicationStarting(CDIRuntimeImpl.java:453)
	at com.ibm.ws.container.service.state.internal.ApplicationStateManager.fireStarting(ApplicationStateManager.java:51)
	at com.ibm.ws.container.service.state.internal.StateChangeServiceImpl.fireApplicationStarting(StateChangeServiceImpl.java:50)
	at com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase.preDeployApp(SimpleDeployedAppInfoBase.java:550)
	at com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase.installApp(SimpleDeployedAppInfoBase.java:511)
	at com.ibm.ws.app.manager.module.internal.DeployedAppInfoBase.deployApp(DeployedAppInfoBase.java:347)
	at com.ibm.ws.app.manager.war.internal.WARApplicationHandlerImpl.install(WARApplicationHandlerImpl.java:65)
	at com.ibm.ws.app.manager.internal.statemachine.StartAction.execute(StartAction.java:140)
	at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.enterState(ApplicationStateMachineImpl.java:1264)
	at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.run(ApplicationStateMachineImpl.java:879)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:821)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory.getInstance(KafkaAdapterFactory.java:165)
	... 55 more
Caused by: org.apache.kafka.common.KafkaException: Failed to construct kafka consumer
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:820)
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:631)
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:612)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.impl.KafkaConsumerImpl.<init>(KafkaConsumerImpl.java:37)
	... 60 more
Caused by: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers
	at org.apache.kafka.clients.ClientUtils.parseAndValidateAddresses(ClientUtils.java:88)
	at org.apache.kafka.clients.ClientUtils.parseAndValidateAddresses(ClientUtils.java:47)
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:735)
	... 63 more
```

**Describe the solution you'd like**
I would like the root message of the InvocationTargetException to be surfaced in console.log

**Describe alternatives you've considered**
1. Don't ever mess up writing a MP Reactive Messaging app ;)
2. Design issue for disabling stack trace hiding in STDOUT: https://github.com/OpenLiberty/open-liberty/issues/10904

--- a/dev/com.ibm.ws.microprofile.reactive.messaging.kafka.adapter/src/com/ibm/ws/microprofile/reactive/messaging/kafka/adapter/KafkaAdapterException.java
+++ b/dev/com.ibm.ws.microprofile.reactive.messaging.kafka.adapter/src/com/ibm/ws/microprofile/reactive/messaging/kafka/adapter/KafkaAdapterException.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2019 IBM Corporation and others.
+ * Copyright (c) 2019, 2020 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -9,6 +9,8 @@
  *     IBM Corporation - initial API and implementation
  *******************************************************************************/
 package com.ibm.ws.microprofile.reactive.messaging.kafka.adapter;
+
+import java.lang.reflect.InvocationTargetException;
 
 /**
  *
@@ -22,7 +24,7 @@ public class KafkaAdapterException extends RuntimeException {
      * @param cause
      */
     public KafkaAdapterException(Exception cause) {
-        super(cause);
+        super(buildMessage(cause), cause);
     }
 
     /**
@@ -30,6 +32,36 @@ public class KafkaAdapterException extends RuntimeException {
      */
     public KafkaAdapterException(String message) {
         super(message);
+    }
+
+    /**
+     * Build a useful message based on the cause chain from Kafka
+     * <p>
+     * In general, the kafka client gives a helpful chain of exceptions explaining what happened and when. However, for liberty log messages, we want to capture the exception cause
+     * chain in a single message.
+     *
+     * @param cause the cause of this exception
+     * @return the message for this exception
+     */
+    private static String buildMessage(Throwable cause) {
+        // Usually the first exception is an InvocationTargetException which isn't helpful in the error message
+        if (cause instanceof InvocationTargetException) {
+            cause = cause.getCause();
+        }
+
+        StringBuilder sb = new StringBuilder();
+        boolean first = true;
+        while (cause != null) {
+            if (!first) {
+                sb.append(": ");
+            }
+            first = false;
+
+            sb.append(cause.toString());
+            cause = cause.getCause();
+        }
+
+        return sb.toString();
     }
 
 }

--- a/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops
+++ b/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops
@@ -60,6 +60,16 @@ kafka.library.not.present.CWMRX1006E=CWMRX1006E: Classes from the kafka-clients 
 kafka.library.not.present.CWMRX1006E.explanation=Applications that use the Reactive Messaging Kafka Client must provide the kafka-clients JAR file and its dependencies. They can provide this file and its dependencies either by packaging it as a library within the application or by providing it as a shared library.
 kafka.library.not.present.CWMRX1006E.useraction=Package the kafka-clients JAR file and its dependencies as libraries in your application or as a shared library. The kafka-clients JAR file can be downloaded from the Apache Kafka website or from Maven Central.
 
+# Could not create a Kafka incoming connector for the {0} channel. The error is {1}
+kafka.create.incoming.error.CWMRX1007E=CWMRX1007E: Could not create a Kafka incoming connector for the {0} channel. The error is {1}
+kafka.create.incoming.error.CWMRX1007E.explanation=An error has occurred while setting up the Kafka connector for an incoming reactive messaging channel that prevents normal operation of Reactive Messaging.
+kafka.create.incoming.error.CWMRX1007E.useraction=Review the error message, server message.log and FFDC logs to identify the problem.
+
+# Could not create a Kafka outgoing connector for the {0} channel. The error is {1}
+kafka.create.outgoing.error.CWMRX1008E=CWMRX1008E: Could not create a Kafka outgoing connector for the {0} channel. The error is {1}
+kafka.create.outgoing.error.CWMRX1008E.explanation=An error has occurred while setting up the Kafka connector for an outgoing reactive messaging channel that prevents normal operation of Reactive Messaging.
+kafka.create.outgoing.error.CWMRX1008E.useraction=Review the error message, server message.log and FFDC logs to identify the problem.
+
 #-----------------------------------------------------------------------------------------------------------------------------
 # Emergency Reactive Messaging error message
 #-----------------------------------------------------------------------------------------------------------------------------

--- a/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops
+++ b/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops
@@ -60,15 +60,15 @@ kafka.library.not.present.CWMRX1006E=CWMRX1006E: Classes from the kafka-clients 
 kafka.library.not.present.CWMRX1006E.explanation=Applications that use the Reactive Messaging Kafka Client must provide the kafka-clients JAR file and its dependencies. They can provide this file and its dependencies either by packaging it as a library within the application or by providing it as a shared library.
 kafka.library.not.present.CWMRX1006E.useraction=Package the kafka-clients JAR file and its dependencies as libraries in your application or as a shared library. The kafka-clients JAR file can be downloaded from the Apache Kafka website or from Maven Central.
 
-# Could not create a Kafka incoming connector for the {0} channel. The error is {1}
-kafka.create.incoming.error.CWMRX1007E=CWMRX1007E: Could not create a Kafka incoming connector for the {0} channel. The error is {1}
-kafka.create.incoming.error.CWMRX1007E.explanation=An error has occurred while setting up the Kafka connector for an incoming reactive messaging channel that prevents normal operation of MicroProfile Reactive Messaging.
-kafka.create.incoming.error.CWMRX1007E.useraction=Review the error message, server message.log and FFDC logs to identify the problem.
+# A Kafka incoming connector for the {0} channel cannot be created. The error is {1}
+kafka.create.incoming.error.CWMRX1007E=CWMRX1007E: A Kafka incoming connector for the {0} channel cannot be created. The error is {1}
+kafka.create.incoming.error.CWMRX1007E.explanation=An error that occurred when the Kafka connector was set up for an incoming reactive messaging channel prevents normal operation of MicroProfile Reactive Messaging.
+kafka.create.incoming.error.CWMRX1007E.useraction=Review the error message, server message.log, and FFDC logs to identify the problem.
 
-# Could not create a Kafka outgoing connector for the {0} channel. The error is {1}
-kafka.create.outgoing.error.CWMRX1008E=CWMRX1008E: Could not create a Kafka outgoing connector for the {0} channel. The error is {1}
-kafka.create.outgoing.error.CWMRX1008E.explanation=An error has occurred while setting up the Kafka connector for an outgoing reactive messaging channel that prevents normal operation of MicroProfile Reactive Messaging.
-kafka.create.outgoing.error.CWMRX1008E.useraction=Review the error message, server message.log and FFDC logs to identify the problem.
+# A Kafka outgoing connector for the {0} channel cannot be created. The error is {1}
+kafka.create.outgoing.error.CWMRX1008E=CWMRX1008E: A Kafka outgoing connector for the {0} channel cannot be created. The error is {1}
+kafka.create.outgoing.error.CWMRX1008E.explanation=An error that occurred when the Kafka connector was set up for an outgoing reactive messaging channel prevents normal operation of MicroProfile Reactive Messaging.
+kafka.create.outgoing.error.CWMRX1008E.useraction=Review the error message, server message.log, and FFDC logs to identify the problem.
 
 #-----------------------------------------------------------------------------------------------------------------------------
 # Emergency Reactive Messaging error message

--- a/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops
+++ b/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops
@@ -62,12 +62,12 @@ kafka.library.not.present.CWMRX1006E.useraction=Package the kafka-clients JAR fi
 
 # Could not create a Kafka incoming connector for the {0} channel. The error is {1}
 kafka.create.incoming.error.CWMRX1007E=CWMRX1007E: Could not create a Kafka incoming connector for the {0} channel. The error is {1}
-kafka.create.incoming.error.CWMRX1007E.explanation=An error has occurred while setting up the Kafka connector for an incoming reactive messaging channel that prevents normal operation of Reactive Messaging.
+kafka.create.incoming.error.CWMRX1007E.explanation=An error has occurred while setting up the Kafka connector for an incoming reactive messaging channel that prevents normal operation of MicroProfile Reactive Messaging.
 kafka.create.incoming.error.CWMRX1007E.useraction=Review the error message, server message.log and FFDC logs to identify the problem.
 
 # Could not create a Kafka outgoing connector for the {0} channel. The error is {1}
 kafka.create.outgoing.error.CWMRX1008E=CWMRX1008E: Could not create a Kafka outgoing connector for the {0} channel. The error is {1}
-kafka.create.outgoing.error.CWMRX1008E.explanation=An error has occurred while setting up the Kafka connector for an outgoing reactive messaging channel that prevents normal operation of Reactive Messaging.
+kafka.create.outgoing.error.CWMRX1008E.explanation=An error has occurred while setting up the Kafka connector for an outgoing reactive messaging channel that prevents normal operation of MicroProfile Reactive Messaging.
 kafka.create.outgoing.error.CWMRX1008E.useraction=Review the error message, server message.log and FFDC logs to identify the problem.
 
 #-----------------------------------------------------------------------------------------------------------------------------

--- a/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaConnectorException.java
+++ b/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaConnectorException.java
@@ -13,7 +13,7 @@ package com.ibm.ws.microprofile.reactive.messaging.kafka;
 /**
  * General exception for problems which originate in the Kafka connector
  */
-public class KafkaConnectorException extends Exception {
+public class KafkaConnectorException extends RuntimeException {
 
     private static final long serialVersionUID = 1L;
 

--- a/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaIncomingConnector.java
+++ b/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaIncomingConnector.java
@@ -86,49 +86,53 @@ public class KafkaIncomingConnector implements IncomingConnectorFactory {
         // Extract our config
         String channelName = config.getValue(ConnectorFactory.CHANNEL_NAME_ATTRIBUTE, String.class);
 
-        Optional<String> groupID = config.getOptionalValue(KafkaConnectorConstants.GROUP_ID, String.class);
-        if (!groupID.isPresent()) {
-            String msg = Tr.formatMessage(tc, "kafka.groupid.not.set.CWMRX1005E", ConnectorFactory.INCOMING_PREFIX + channelName + "." + KafkaConnectorConstants.GROUP_ID);
-            throw new IllegalArgumentException(msg);
+        try {
+            Optional<String> groupID = config.getOptionalValue(KafkaConnectorConstants.GROUP_ID, String.class);
+            if (!groupID.isPresent()) {
+                String msg = Tr.formatMessage(tc, "kafka.groupid.not.set.CWMRX1005E", ConnectorFactory.INCOMING_PREFIX + channelName + "." + KafkaConnectorConstants.GROUP_ID);
+                throw new IllegalArgumentException(msg);
+            }
+
+            String topic = config.getOptionalValue(KafkaConnectorConstants.TOPIC, String.class).orElse(channelName);
+            int maxPollRecords = config.getOptionalValue(KafkaConnectorConstants.MAX_POLL_RECORDS, Integer.class).orElse(500);
+            int unackedLimit = config.getOptionalValue(KafkaConnectorConstants.UNACKED_LIMIT, Integer.class).orElse(maxPollRecords);
+
+            // Configure our defaults
+            Map<String, Object> consumerConfig = new HashMap<>();
+            // Default behaviour is that connector handles commit in response to ack()
+            consumerConfig.put(KafkaConnectorConstants.ENABLE_AUTO_COMMIT, "false");
+            //default the key and value deserializers to String
+            consumerConfig.put(KafkaConnectorConstants.KEY_DESERIALIZER, KafkaConnectorConstants.STRING_DESERIALIZER);
+            consumerConfig.put(KafkaConnectorConstants.VALUE_DESERIALIZER, KafkaConnectorConstants.STRING_DESERIALIZER);
+
+            // Pass the rest of the config directly through to the kafkaConsumer
+            consumerConfig.putAll(StreamSupport.stream(config.getPropertyNames().spliterator(), false)
+                                               .filter(k -> !KafkaConnectorConstants.NON_KAFKA_PROPS.contains(k))
+                                               .collect(Collectors.toMap(Function.identity(), (k) -> config.getValue(k, String.class))));
+
+            boolean enableAutoCommit = "true".equalsIgnoreCase((String) consumerConfig.get(KafkaConnectorConstants.ENABLE_AUTO_COMMIT));
+
+            // Create the kafkaConsumer
+            KafkaConsumer<String, Object> kafkaConsumer = this.kafkaAdapterFactory.newKafkaConsumer(consumerConfig);
+
+            PartitionTrackerFactory partitionTrackerFactory = new PartitionTrackerFactory();
+            partitionTrackerFactory.setExecutor(executor);
+            partitionTrackerFactory.setAdapterFactory(kafkaAdapterFactory);
+            partitionTrackerFactory.setAutoCommitEnabled(enableAutoCommit);
+
+            if (enableAutoCommit) {
+                unackedLimit = 0;
+            }
+
+            // Create our connector around the kafkaConsumer
+            KafkaInput<String, Object> kafkaInput = new KafkaInput<>(this.kafkaAdapterFactory, partitionTrackerFactory, kafkaConsumer, this.executor,
+                                                                     topic, unackedLimit);
+            kafkaInputs.add(kafkaInput);
+
+            return kafkaInput.getPublisher();
+        } catch (Exception e) {
+            throw new KafkaConnectorException(Tr.formatMessage(tc, "kafka.create.incoming.error.CWMRX1007E", channelName, e.getMessage()), e);
         }
-
-        String topic = config.getOptionalValue(KafkaConnectorConstants.TOPIC, String.class).orElse(channelName);
-        int maxPollRecords = config.getOptionalValue(KafkaConnectorConstants.MAX_POLL_RECORDS, Integer.class).orElse(500);
-        int unackedLimit = config.getOptionalValue(KafkaConnectorConstants.UNACKED_LIMIT, Integer.class).orElse(maxPollRecords);
-
-        // Configure our defaults
-        Map<String, Object> consumerConfig = new HashMap<>();
-        // Default behaviour is that connector handles commit in response to ack()
-        consumerConfig.put(KafkaConnectorConstants.ENABLE_AUTO_COMMIT, "false");
-        //default the key and value deserializers to String
-        consumerConfig.put(KafkaConnectorConstants.KEY_DESERIALIZER, KafkaConnectorConstants.STRING_DESERIALIZER);
-        consumerConfig.put(KafkaConnectorConstants.VALUE_DESERIALIZER, KafkaConnectorConstants.STRING_DESERIALIZER);
-
-        // Pass the rest of the config directly through to the kafkaConsumer
-        consumerConfig.putAll(StreamSupport.stream(config.getPropertyNames().spliterator(), false)
-                                           .filter(k -> !KafkaConnectorConstants.NON_KAFKA_PROPS.contains(k))
-                                           .collect(Collectors.toMap(Function.identity(), (k) -> config.getValue(k, String.class))));
-
-        boolean enableAutoCommit = "true".equalsIgnoreCase((String) consumerConfig.get(KafkaConnectorConstants.ENABLE_AUTO_COMMIT));
-
-        // Create the kafkaConsumer
-        KafkaConsumer<String, Object> kafkaConsumer = this.kafkaAdapterFactory.newKafkaConsumer(consumerConfig);
-
-        PartitionTrackerFactory partitionTrackerFactory = new PartitionTrackerFactory();
-        partitionTrackerFactory.setExecutor(executor);
-        partitionTrackerFactory.setAdapterFactory(kafkaAdapterFactory);
-        partitionTrackerFactory.setAutoCommitEnabled(enableAutoCommit);
-
-        if (enableAutoCommit) {
-            unackedLimit = 0;
-        }
-
-        // Create our connector around the kafkaConsumer
-        KafkaInput<String, Object> kafkaInput = new KafkaInput<>(this.kafkaAdapterFactory, partitionTrackerFactory, kafkaConsumer, this.executor,
-                                                                 topic, unackedLimit);
-        kafkaInputs.add(kafkaInput);
-
-        return kafkaInput.getPublisher();
     }
 
 }

--- a/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaOutgoingConnector.java
+++ b/dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/KafkaOutgoingConnector.java
@@ -32,12 +32,16 @@ import org.eclipse.microprofile.reactive.messaging.spi.OutgoingConnectorFactory;
 import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
 import org.eclipse.microprofile.reactive.streams.operators.SubscriberBuilder;
 
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
 import com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory;
 import com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaProducer;
 
 @Connector(KafkaConnectorConstants.CONNECTOR_NAME)
 @ApplicationScoped
 public class KafkaOutgoingConnector implements OutgoingConnectorFactory {
+
+    private static final TraceComponent tc = Tr.register(KafkaOutgoingConnector.class);
 
     @Inject
     private KafkaAdapterFactory kafkaAdapterFactory;
@@ -46,27 +50,33 @@ public class KafkaOutgoingConnector implements OutgoingConnectorFactory {
 
     @Override
     public SubscriberBuilder<Message<Object>, Void> getSubscriberBuilder(Config config) {
-        // Configure our defaults
-        Map<String, Object> producerConfig = new HashMap<>();
-
-        //default the key and value serializers to String
-        producerConfig.put(KafkaConnectorConstants.KEY_SERIALIZER, KafkaConnectorConstants.STRING_SERIALIZER);
-        producerConfig.put(KafkaConnectorConstants.VALUE_SERIALIZER, KafkaConnectorConstants.STRING_SERIALIZER);
-
-        // Pass the config directly through to the kafkaProducer
-        producerConfig.putAll(StreamSupport.stream(config.getPropertyNames().spliterator(), false)
-                                           .filter(k -> !KafkaConnectorConstants.NON_KAFKA_PROPS.contains(k))
-                                           .collect(Collectors.toMap(Function.identity(), (k) -> config.getValue(k, String.class))));
-
-        KafkaProducer<String, Object> kafkaProducer = this.kafkaAdapterFactory.newKafkaProducer(producerConfig);
 
         String channelName = config.getValue(ConnectorFactory.CHANNEL_NAME_ATTRIBUTE, String.class);
-        String configuredTopic = config.getOptionalValue(KafkaConnectorConstants.TOPIC, String.class).orElse(null);
 
-        KafkaOutput<String, Object> kafkaOutput = new KafkaOutput<>(this.kafkaAdapterFactory, configuredTopic, channelName, kafkaProducer);
-        this.kafkaOutputs.add(kafkaOutput);
+        try {
+            // Configure our defaults
+            Map<String, Object> producerConfig = new HashMap<>();
 
-        return ReactiveStreams.<Message<Object>> builder().to(kafkaOutput.getSubscriber());
+            //default the key and value serializers to String
+            producerConfig.put(KafkaConnectorConstants.KEY_SERIALIZER, KafkaConnectorConstants.STRING_SERIALIZER);
+            producerConfig.put(KafkaConnectorConstants.VALUE_SERIALIZER, KafkaConnectorConstants.STRING_SERIALIZER);
+
+            // Pass the config directly through to the kafkaProducer
+            producerConfig.putAll(StreamSupport.stream(config.getPropertyNames().spliterator(), false)
+                                               .filter(k -> !KafkaConnectorConstants.NON_KAFKA_PROPS.contains(k))
+                                               .collect(Collectors.toMap(Function.identity(), (k) -> config.getValue(k, String.class))));
+
+            KafkaProducer<String, Object> kafkaProducer = this.kafkaAdapterFactory.newKafkaProducer(producerConfig);
+
+            String configuredTopic = config.getOptionalValue(KafkaConnectorConstants.TOPIC, String.class).orElse(null);
+
+            KafkaOutput<String, Object> kafkaOutput = new KafkaOutput<>(this.kafkaAdapterFactory, configuredTopic, channelName, kafkaProducer);
+            this.kafkaOutputs.add(kafkaOutput);
+
+            return ReactiveStreams.<Message<Object>> builder().to(kafkaOutput.getSubscriber());
+        } catch (Exception e) {
+            throw new KafkaConnectorException(Tr.formatMessage(tc, "kafka.create.outgoing.error.CWMRX1008E", channelName, e.getMessage()), e);
+        }
     }
 
     @PreDestroy

